### PR TITLE
Fixed broken dev-env

### PR DESCRIPTION
**Pull request by salvatore-coppola**
**Fixed broken dev-env**

Modified dependencies version, launch configurations (for both OSX and linux), initial emulator snapshot and bundles composition on emulator target platform.

Open points:

- An `java.lang.IllegalStateException` is still thrown because the default clientId (primal network interface mac address) in MqttDataTransport could not be read if the primal network interface is not eth0. Not blocking but we could set a default string as clientId.
- A `java.lang.IllegalArgumentException` is thrown at equinox startup because standard property `osgi.configuration.area` is overridden by `org.osgi.framework.storage` (not a real issue).
- A `javax.management.InstanceAlreadyExistsException` is thrown by `log4j2` mbean (not an issue).

### CODE REVIEW

**Comment by salvatore-coppola** [comment]
- Add a random generated clientId to the emulator's snapshot_0.

**Comment by MMaiero** [comment]
Can you check the sonar results?

**Comment by salvatore-coppola** [comment]
@MMaiero currently there isn't test coverage for the emulator bundle, should I add it ?

File: kura/emulator/org.eclipse.kura.emulator/src/main/java/org/eclipse/kura/emulator/Emulator.java [hunk 1]
```diff
@@ -66,23 +74,23 @@ protected void deactivate(ComponentContext componentContext) {
...
-            if (fileOutput != null) {
-                fileOutput.close();
-            }
-            if (fileInput != null) {
-                fileInput.close();
-            }
+        URL internalSnapshotURL = this.componentContext.getBundleContext().getBundle().getResource(SNAPSHOT_0_NAME);
+
+        try (InputStream fileInput = internalSnapshotURL.openStream();
+                OutputStream fileOutput = new FileOutputStream(snapshotFolderPath + File.separator + SNAPSHOT_0_NAME)) {
```
Comment (MMaiero) [inline kura/emulator/org.eclipse.kura.emulator/src/main/java/org/eclipse/kura/emulator/Emulator.java:80]: Formatting seems not applied properly

File: target-platform/config/kura.target-platform.build.properties [hunk 2]
```diff
@@ -7,7 +7,7 @@ org.usb4java.version=1.2.500
 usb4java-javax.version=1.2.500
 com.gwt.user.version=1.1.100
 jdk.dio.version=1.0.500
-org.eclipse.kura.camel.sun.misc.version=1.0.400
+org.eclipse.kura.camel.sun.misc.version=1.0.500-SNAPSHOT
```
Comment (MMaiero) [inline target-platform/config/kura.target-platform.build.properties:10]: Is this change required?

**Comment by MMaiero** [comment]
In my setup the dev-env is starting but the system packages section is not loading. Probably because it is searching for RPMs?

File: target-platform/config/kura.target-platform.build.properties (hunk 2 above)
Comment (salvatore-coppola) [reply target-platform/config/kura.target-platform.build.properties:10]: I checked and yes we need it

**Review by MMaiero — APPROVED** [review]
